src/Http/Routing/test/UnitTests/EndpointMiddlewareTest.cs

```diff
@@ -314,11 +315,115 @@ public async Task Invoke_WithEndpoint_DoesNotThrowIfUnhandledCorsAttributesWereF
         Assert.True(calledEndpoint);
     }
 
+    [Fact]
+    public async Task Invoke_WithEndpoint_ThrowsIfAntiforgeryMetadataWasFound_ButAntiforgeryMiddlewareNotInvoked()
+    {
+        // Arrange
+        var expected = "Endpoint Test contains anti-forgery metadata, but a middleware was not found that supports anti-forgery." +
+            Environment.NewLine +
+            "Configure your application startup by adding app.UseAntiforgery() in the application startup code. " +
+            "If there are calls to app.UseRouting() and app.UseEndpoints(...), the call to app.UseAntiforgery() must go between them.";
+        var httpContext = new DefaultHttpContext
+        {
+            RequestServices = new ServiceProvider()
+        };
+
+        RequestDelegate throwIfCalled = (c) =>
+        {
+            throw new InvalidTimeZoneException("Should not be called");
+        };
+
+        httpContext.SetEndpoint(new Endpoint(throwIfCalled, new EndpointMetadataCollection(new AntiforgeryMetadata(true)), "Test"));
+
+        var middleware = new EndpointMiddleware(NullLogger<EndpointMiddleware>.Instance, throwIfCalled, RouteOptions);
+
+        // Act & Assert
+        var ex = await Assert.ThrowsAsync<InvalidOperationException>(() => middleware.Invoke(httpContext));
```

Comment:
Do we need to filter for the exception message here so that we don't get a false pass.

Reply:
We filter for it in the next check so this should be good.